components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,955 @@
+module scream_scorpio_interface
+
+!==============================================================================!
+! This module handles the Fortran interface to the PIO library of input/output
+! subroutines.  The essential set of steps to enable and use PIO for creating
+! output are as follows:
+!                                  OUTPUT
+! Initialization:
+! 1) Gather the pio_subsystem and pio_iotype information for the EAM component
+! as assigned by the component coupler.
+!    This is accomplished during eam_init_pio by calling
+!    'eam_init_pio_subsystem'
+! 2) For each output file "create" a file in PIO and record the unique file
+! descriptor. 
+!    This is accomplished during the eam_init_pio by calling
+!    'eam_pio_createfile'
+! 3) For each output file define the "header" information, which is essentially
+! a set of metadata strings that describe the output file.
+!    This is accomplished during eam_init_pio by calling 'eam_pio_createHeader'
+! 4) Define all of the dimensions that variables in this file will be defined
+! on.  Examples would time, lat, lon, vertical coordinate, # of consituents,
+! etc.
+!    This is accomplished during 'h_define' by repeated calls to 'PIO_def_dim'
+! 5) Define all of the variables that will be written to this file.  This
```

Comment:
h_define isn't a great name for a file/function. What does h mean?

Reply:
great catch, h_define was the old name from CAM, I didn't update the comments section when I decided to use the name `register_variable`